editor/src/uuiui/inputs/string-input.spec.browser2.tsx

```diff
@@ -77,4 +83,106 @@ describe('StringInput', () => {
+          editorStoreData={storeHookForTest}
+        >
+          <StringInput testId='the-input' value='hello' />
+        </TestInspectorContextProvider>,
+      )
+
+      const inputElement = await result.findByTestId('the-input')
+      if (inputElement == null || !(inputElement instanceof HTMLInputElement)) {
+        throw new Error('Could not find input element.')
+      }
+
+      firePasteTextEvent(inputElement, 'well hello there!')
+      await wait(100)
+
+      expect(inputElement.value).toBe('hello')
+    })
+
+    it('does nothing if the pasted text is utopia data', async () => {
+      const storeHookForTest = getStoreHook()
+      const result = render(
+        <TestInspectorContextProvider
+          selectedViews={storeHookForTest.getState().editor.selectedViews}
+          editorStoreData={storeHookForTest}
+        >
+          <StringInputWithChangeHandler testId='the-input' value='hello' />
+        </TestInspectorContextProvider>,
+      )
+
+      const inputElement = await result.findByTestId('the-input')
+      if (inputElement == null || !(inputElement instanceof HTMLInputElement)) {
+        throw new Error('Could not find input element.')
+      }
+
+      const imagesToPaste = [await makeImageFile(imgBase641x1, 'chucknorris.png')]
+      firePasteImageEvent(inputElement, imagesToPaste)
+      await wait(100)
+
+      expect(inputElement.value).toBe('hello')
+    })
+
+    it('does nothing if the pasted text is empty', async () => {
+      const storeHookForTest = getStoreHook()
+      const result = render(
+        <TestInspectorContextProvider
+          selectedViews={storeHookForTest.getState().editor.selectedViews}
+          editorStoreData={storeHookForTest}
+        >
+          <StringInputWithChangeHandler testId='the-input' value='hello' />
+        </TestInspectorContextProvider>,
+      )
+
+      const inputElement = await result.findByTestId('the-input')
+      if (inputElement == null || !(inputElement instanceof HTMLInputElement)) {
+        throw new Error('Could not find input element.')
+      }
+
+      firePasteTextEvent(inputElement, '')
+      await wait(100)
+
+      expect(inputElement.value).toBe('hello')
+    })
+
+    it('accepts pasted text', async () => {
+      const storeHookForTest = getStoreHook()
+      const result = render(
+        <TestInspectorContextProvider
+          selectedViews={storeHookForTest.getState().editor.selectedViews}
+          editorStoreData={storeHookForTest}
+        >
+          <StringInputWithChangeHandler testId='the-input' value='hello' />
+        </TestInspectorContextProvider>,
+      )
+
+      const inputElement = await result.findByTestId('the-input')
+      if (inputElement == null || !(inputElement instanceof HTMLInputElement)) {
+        throw new Error('Could not find input element.')
+      }
+
+      firePasteTextEvent(inputElement, 'well hello there!')
+      await wait(100)
```

Comment:
could this be replaced by an `await firePasteTextEvent`? if not, then `await result.getDispatchFollowUpActionsFinished()` ? 

the await(0) is sometimes acceptable as it means we wait a tick. but awaiting a concrete number of milliseconds should never be used because it means there is possibility for the test to become flaky

Reply:
the render result is not a `EditorRenderResult`, so I just changed the awaits to be 0s